pymc3/distributions/logprob.py

```diff
@@ -141,8 +129,89 @@ def logpt(
         A scaling term to apply to the generated log-likelihood graph.
     transformed
         Apply transforms.
-    cdf
-        Return the log cumulative distribution.
+
+    """
+    if not isinstance(rv_values, Mapping):
+        rv_values = (
+            {var: at.as_tensor_variable(rv_values).astype(var.type)}
+            if rv_values is not None
+            else {}
+        )
+
+    tmp_rv_values = {}
+    for node in io_toposort(graph_inputs((var,)), (var,)):
+        if isinstance(node.op, RandomVariable):
+            tmp_rv_values[node.out] = node.out
+
+    logp_var = joint_logprob(var, tmp_rv_values, **kwargs)
+
+    rv_value_var = getattr(var.tag, "observations", getattr(var.tag, "value_var", None))
+    transform = getattr(rv_value_var.tag, "transform", None) if rv_value_var else None
+    rv_value = rv_values.get(var, rv_value_var)
+
+    if transform and transformed and jacobian:
+        transformed_jacobian = transform.jacobian_det(var, rv_value)
+        if transformed_jacobian:
+            if logp_var.ndim > transformed_jacobian.ndim:
+                logp_var = logp_var.sum(axis=-1)
+            logp_var += transformed_jacobian
+
+    # Replace random variables with their value variables
+    replacements = rv_values.copy()
+    replacements.update({var: rv_value})
+
+    (logp_var,), _ = rvs_to_value_vars(
+        (logp_var,),
+        apply_transforms=transformed,
+        initial_replacements=replacements,
+    )
+
+    if sum:
+        logp_var = at.sum(logp_var)
+
+    if scaling:
+        logp_var *= _get_scaling(
+            getattr(var.tag, "total_size", None), rv_value.shape, rv_value.ndim
+        )
+
+    # Recompute test values for the changes introduced by the replacements
+    # above.
+    if config.compute_test_value != "off":
+        for node in io_toposort(graph_inputs((logp_var,)), (logp_var,)):
+            compute_test_value(node)
+
+    if var.name is not None:
+        logp_var.name = "__logp_%s" % var.name
+
+    return logp_var
+
+
+def logcdfpt(
```

Comment:
Separated the `logcdf` and `logp` frameworks since down the line we can expect changes to `logpt` function when `joint_logprob` fully supports transforms. (And the difference between the logic for handling one thing or other will increase)